tpot/tpot.py

```diff
@@ -1135,6 +1141,210 @@ def _rbf(self, input_df, gamma, n_components):
+        gamma: float
+            Gamma parameter for the kernels.
+        n_components: int
+            The number of components to keep
+
+        Returns
+        -------
+        modified_df: pandas.DataFrame {n_samples, n_components + ['guess', 'group', 'class']}
+            Returns a DataFrame containing the transformed features
+
+        """
+        training_features = input_df.loc[input_df['group'] == 'training'].drop(self.non_feature_columns, axis=1)
+
+        if len(training_features.columns.values) == 0:
+            return input_df.copy()
+
+        if n_components < 1:
+            n_components = 1
+        else:
+            n_components = min(n_components, len(training_features.columns.values))
+
+        kernel_types = list(PAIRWISE_KERNEL_FUNCTIONS.keys())
+
+        kernel_name = kernel_types[kernel % len(kernel_types)]
+
+        nys = Nystroem(kernel=kernel_name, gamma=gamma, n_components=n_components)
+        nys.fit(training_features.values.astype(np.float64))
+        transformed_features = nys.transform(input_df.drop(self.non_feature_columns, axis=1).values.astype(np.float64))
+
+        modified_df = pd.DataFrame(data=transformed_features)
+        modified_df['class'] = input_df['class'].values
+        modified_df['group'] = input_df['group'].values
+        modified_df['guess'] = input_df['guess'].values
+
+        new_col_names = {}
+        for column in modified_df.columns.values:
+            if type(column) != str:
+                new_col_names[column] = str(column).zfill(10)
+        modified_df.rename(columns=new_col_names, inplace=True)
+
+        return modified_df.copy()
+
+    def _zero_count(self, input_df):
+        """
+        Adds virtual features for the number of zeros per row, and number of
+        non-zeros per row.
+
+        Parameters
+        ----------
+        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
+            Input DataFrame to scale
+
+        Returns
+        -------
+        modified_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
+            Returns a DataFrame containing the new virtual features
+
+        """
+        feature_cols_only = input_df.drop(self.non_feature_columns, axis=1)
+
+        num_features = len(feature_cols_only.columns.values)
+
+        if num_features == 0:
+            return input_df.copy()
+
+        non_zero_col = np.array([np.count_nonzero(row) for i, row in feature_cols_only.iterrows()]).astype(np.float64)
+        zero_col     = np.array([(num_features - x) for x in non_zero_col]).astype(np.float64)
+
+        modified_df = feature_cols_only.copy()
+        modified_df['non_zero'] = pd.Series(non_zero_col, index=modified_df.index)
+        modified_df['zero_col'] = pd.Series(zero_col, index=modified_df.index)
+
+        modified_df['class'] = input_df['class'].values
+        modified_df['group'] = input_df['group'].values
+        modified_df['guess'] = input_df['guess'].values
```

Comment:
Well the class column is a float (after data cleaning/preprocessing), right? I don't want to count it as a column with a value of zero/non-zero.